src/pages/submit/remove-poi.tsx

```diff
@@ -0,0 +1,257 @@
+import Label from "decentraland-gatsby/dist/components/Form/Label"
+import useFormatMessage from "decentraland-gatsby/dist/hooks/useFormatMessage"
+import useEditor, {
+  assert,
+  createValidator,
+} from "decentraland-gatsby/dist/hooks/useEditor"
+import ContentLayout, {
+  ContentSection,
+} from "../../components/Layout/ContentLayout"
+import { Governance } from "../../api/Governance"
+import locations from "../../modules/locations"
+import loader from "../../modules/loader"
+import useAuthContext from "decentraland-gatsby/dist/context/Auth/useAuthContext"
+import Head from "decentraland-gatsby/dist/components/Head/Head"
+import MarkdownNotice from "../../components/Form/MarkdownNotice"
+import "./submit.css"
+
+type POIState = {
+  x: string | number
+  y: string | number
+  description: string
+}
+
+const schema = newProposalPOIScheme.properties
+const initialPollState: POIState = {
+  x: "",
+  y: "",
+  description: "",
+}
+
+const edit = (state: POIState, props: Partial<POIState>) => {
+  return {
+    ...state,
+    ...props,
+  }
+}
+
+function assertInMap(value: string | number) {
+  if (value === "") {
+    return undefined
+  }
+
+  const x = asNumber(value)
+  return (
+    assert(
+      !Number.isNaN(x) && Number.isFinite(x),
+      "error.poi.coordinates_not_a_number"
+    ) ||
+    assert(
+      x <= schema.x.maximum && x >= schema.x.minimum,
+      "error.poi.coordinates_out_of_map"
+    ) ||
+    undefined
+  )
+}
+
+const validate = createValidator<POIState>({
+  x: (state) => ({ x: assertInMap(state.x) }),
+  y: (state) => ({ y: assertInMap(state.y) }),
+  description: (state) => ({
+    description: assert(
+      state.description.length <= schema.description.maxLength,
+      "error.poi.description_too_large"
+    ),
+  }),
+  "*": (state) => ({
+    x:
+      assert(state.x !== "", "error.poi.coordinates_incomplete") ||
+      assertInMap(state.x),
+    y:
+      assert(state.y !== "", "error.poi.coordinates_incomplete") ||
+      assertInMap(state.y),
+    description:
+      assert(state.description.length > 0, "error.poi.description_empty") ||
+      assert(
+        state.description.length >= schema.description.minLength,
+        "error.poi.description_too_short"
+      ) ||
+      assert(
+        state.description.length <= schema.description.maxLength,
+        "error.poi.description_too_large"
+      ),
+  }),
+})
+
+export default function SubmitPOI() {
+  const l = useFormatMessage()
+  const [account, accountState] = useAuthContext()
+  const [state, editor] = useEditor(edit, validate, initialPollState)
+
+  useEffect(() => {
+    if (state.validated) {
+      Promise.resolve()
+        .then(async () => {
+          const x = asNumber(state.value.x)
+          const y = asNumber(state.value.y)
+
+          if (!(await isAlreadyPointOfInterest(x, y))) {
+            throw new Error(`error.poi.coordinates_are_not_a_poi`)
+          }
+
+          return Governance.get().createRemoveProposalPOI({
```

Comment:
All this validations and fields in common make me think if we should have only one component for adding/removing, and depending on a url param, change the displayed text and behavior 

For example, depending on the type we could change the validations (as in the back) and do

```js
return Governance.get().createNewProposalPoi({
  x,
  y,
  description: state.value.description,
  poiType: poiType
 })
```
What do you think @yemel ?

Reply:
Thank you very much for all the feedback! Yes, before submitting the PR, I considered refactoring the POI page to a component given the amount of shared functionality. But I was a bit intimidated by the amount of code in the file and I was afraid of breaking something just when I was finishing.

Taking your suggestions, I'll make the changes and resubmit the PR.

Thanks again!